amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/WorkerStateChangeListener.java

```diff
@@ -27,4 +26,6 @@
 	}
 
 	void onWorkerStateChange(WorkerState newState);
+
+	void onInitializationError(Exception e);
```

Comment:
This is a breaking change, and must be defaulted.

I also want to think more about how state reporting and actions happen.  Finally this should accept a `Throwable` vs an `Exception`.